Dockerfile

```diff
@@ -0,0 +1,50 @@
+# Use the official Go image as the base image
+FROM golang:1.21-alpine AS builder
+
+# Version to build. Default is the Git HEAD.
+ARG VERSION="HEAD"
+# Use muslc for static libs
+ARG BUILD_TAGS="muslc"
+
+RUN apk add --no-cache --update openssh git make build-base linux-headers \
+    pkgconfig zeromq-dev libsodium-dev \
+    libzmq-static libsodium-static gcc
+
+RUN mkdir -p /root/.ssh && ssh-keyscan github.com >> /root/.ssh/known_hosts
+RUN git config --global url."git@github.com:".insteadOf "https://github.com/"
+ENV GOPRIVATE=github.com/babylonlabs-io/*
+
+# Build
+WORKDIR /go/src/github.com/babylonlabs-io/finality-gadget
+# Cache dependencies
+COPY go.mod go.sum /go/src/github.com/babylonlabs-io/finality-gadget/
+RUN --mount=type=secret,id=sshKey,target=/root/.ssh/id_rsa go mod download
```

Comment:
this requires ssh key, same issue we met before. let's add similar instructions to README following babylon-integration-deployment repo


```
$ make build-docker                                                    [12:00:56]
/usr/local/bin/docker build --secret id=sshKey,src= \
	--tag babylonlabs-io/finality-gadget \
	-f Dockerfile \
	/Users/zidong/Documents/Projects/babylon-finality-gadget
[+] Building 0.0s (0/0)                                                          docker:desktop-linux
ERROR: failed to stat sshKey: stat sshKey: no such file or directory
make: *** [build-docker] Error 1
```